src/jarabe/desktop/schoolserver.py

```diff
@@ -80,9 +103,26 @@ def _store_identifiers(serial_number, uuid_, backup_url):
     if os.path.exists(os.path.join(identifier_path, 'backup_url')):
         os.remove(os.path.join(identifier_path, 'backup_url'))
     backup_url_file = open(os.path.join(identifier_path, 'backup_url'), 'w')
-    backup_url_file.write(backup_url)
+    backup_url_file.write(jabber_server)
     backup_url_file.close()
 
+    file_path = os.path.join(identifier_path, 'server_history')
+    if os.path.exists(file_path):
+        server_history_file = open(file_path, "r")
+        data = json.load(server_history_file)
+        server_history_file.close()
+        data[serial_number] = {}
+        data[serial_number]["uuid"] = uuid_
+        data[serial_number]["backup_url"] = backup_url
+    else:
```

Comment:
These three lines are identical to the last three lines on the `else` branch of the `if`, please move them out of the conditional clauses.